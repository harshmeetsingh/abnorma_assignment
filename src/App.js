import "./App.css";
import { QueryClient, QueryClientProvider, useQuery } from "react-query";

import Dashboard from "./containers/Dashboard";

const queryClient = new QueryClient();

function App() {
  return (
    <div className="my-App">
      <QueryClientProvider client={queryClient}>
        <Dashboard />
      </QueryClientProvider>
    </div>
  );
}

export default App;
