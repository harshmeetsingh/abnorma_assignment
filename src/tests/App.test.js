import React from "react";
import App from "../App";
import { cleanup, render } from "@testing-library/react";

afterEach(cleanup);

it("Take a snapshot", () => {
  const { asFragment } = render(<App />);
  expect(asFragment(<App />)).toMatchSnapshot();
});
