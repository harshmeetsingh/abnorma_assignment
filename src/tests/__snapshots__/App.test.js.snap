// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Take a snapshot 1`] = `
<DocumentFragment>
  <div
    class="my-App"
  >
    <div
      class="dashboard-container"
    >
      <div
        class="dropdown-container"
      >
        <div
          class="dotted-circular-loader"
        >
          <div
            class="dot"
          />
          <div
            class="dot"
          />
          <div
            class="dot"
          />
        </div>
      </div>
      <div
        class="card-container"
      >
        <div
          class="data-card"
        >
          <div
            class="centre-content data-value"
            style="color: rgb(240, 0, 15);"
          >
            -
          </div>
          <div
            class="centre-content data-label"
          >
            High Severity Attacks
          </div>
        </div>
      </div>
      <div
        class="card-container"
      >
        <div
          class="data-card"
        >
          <div
            class="centre-content data-value"
            style="color: rgb(232, 112, 0);"
          >
            -
          </div>
          <div
            class="centre-content data-label"
          >
            Spam Attacks
          </div>
        </div>
      </div>
      <div
        class="table-container"
      >
        <div
          class="centre-div"
        >
          <table
            class="data-table"
          >
            <thead>
              <tr>
                <th>
                  Top 5 malicious Websites:
                </th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <th>
                  Domain
                </th>
                <th
                  class="data-table-threat-metrics"
                >
                  % of threats
                </th>
                <th
                  class="data-table-threat-metrics"
                >
                  # of threats
                </th>
              </tr>
              <tr>
                <td>
                  Please select a customer
                </td>
              </tr>
            </tbody>
          </table>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
